extmod/modnetwork.h

```diff
@@ -38,6 +38,20 @@
 #define MOD_NETWORK_STA_IF (0)
 #define MOD_NETWORK_AP_IF (1)
 
+// Socket level option.
+#define MOD_NETWORK_SOL_SOCKET      (0x0FFF)
+
+// Common option flags per-socket.
+#define MOD_NETWORK_SO_DEBUG        (0x0001)
+#define MOD_NETWORK_SO_ACCEPTCONN   (0x0002)
+#define MOD_NETWORK_SO_REUSEADDR    (0x0004)
+#define MOD_NETWORK_SO_KEEPALIVE    (0x0008)
+#define MOD_NETWORK_SO_BROADCAST    (0x0020)
+#define MOD_NETWORK_SO_RCVBUF       (0x1002)
+#define MOD_NETWORK_SO_SNDTIMEO     (0x1005)
+#define MOD_NETWORK_SO_RCVTIMEO     (0x1006)
+#define MOD_NETWORK_SO_NO_CHECK     (0x100a)
```

Comment:
I'm not sure all these constants are needed (yet).  `modlwip.c` only provides `SOL_SOCKET` and `SO_REUSEADDR` and those seem to be enough so far.